src/ca/mcgill/cs/jetuml/gui/tips/TipLoader.java

```diff
@@ -1,16 +1,145 @@
 package ca.mcgill.cs.jetuml.gui.tips;
 
+import java.io.InputStream;
+import java.io.InputStreamReader;
 import java.util.ArrayList;
 import java.util.List;
 import java.util.Map;
 
 import org.json.JSONObject;
+import org.json.JSONTokener;
+
+import ca.mcgill.cs.jetuml.application.UserPreferences;
+
+import static ca.mcgill.cs.jetuml.application.ApplicationResources.RESOURCES;
 
 /**
  * Class that statically loads the tips.
  */
 final class TipLoader
 {
+	public static final int NUM_TIPS = Integer.parseInt(RESOURCES.getString("tips.quantity"));
+	private static final String TIP_FILE_PATH_FORMAT = RESOURCES.getString("tips.jsons.directory") + "/tip-%d.json";
+	
+	private TipLoader(){}
+	
+	/**
+	 * Returns the tip associated with the given tip id.
+	 * 
+	 * @param pId id of the tip to return
+	 * @return the tip with id pId.
+	 * 
+	 * @pre pId >= 1 && pId <= NUM_TIPS 
+	 */
+	public static Tip loadTip(int pId)
+	{
+		assert pId >= 1 && pId <= NUM_TIPS;
+		
+		String tipFileName = String.format(TIP_FILE_PATH_FORMAT, pId);
+		InputStream tipsInputStream = TipLoader.class.getResourceAsStream(tipFileName); //Not null if tests pass
+		InputStreamReader tipsReader = new InputStreamReader(tipsInputStream);
+		JSONTokener jsonTokener = new JSONTokener(tipsReader);
+		JSONObject jsonObject = new JSONObject(jsonTokener); // Won't throw JSONException if tests pass
+		Tip tip = new Tip(pId, jsonObject);
+		return tip;
+	}
+	
+	/**
+	 * Method that returns the tip of the day and updates the user preferences 
+	 * to set the user's next tip of the day's id to the id of the following tip
+	 * (i.e. it updates the tip of the day for the next time this method gets 
+	 * called).
+	 * 
+	 * @return Tip the tip of the day
+	 */
+	public static Tip loadTipOfTheDay()
+	{	
+		int tipId = UserPreferences.instance().getInteger(UserPreferences.IntegerPreference.nextTipId);

```

Comment:
I would decouple the management of the user preferences from the TipLoading. Let's put the user preferences stuff in the TipDialog, because conceptually it's when a user "sees" the tip that it matters we store it in the preferences